### PR TITLE
Table translation updates

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table story](https://deploy-preview-463--neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

**Summary**: Updated dropdown translations, added "Clear all" option to the dropdown.

`@avaya-dux/dux-design`

- confirm that the translations do not have any spelling errors
- confirm that the select and clear options work as expected
- confirm that the proper select and clear options show as appropriate based on if there are multiple pages or a single page

`@avaya-dux/dux-devs`

- same as design
- check for any coding mistakes

**EXAMPLE FUNCTIONALITY GIF:**

## Summary by CodeRabbit

- **New Features**
  - Introduced an option for users to select the number of items displayed per page in the data table.
  - Enhanced the `TableHeader` component for improved user interaction with row selections.

- **Bug Fixes**
  - Improved clarity in the user interface by updating selection text to provide better context.

- **Chores**
  - Updated localization strings for better clarity and brevity in user prompts.

- **Documentation**
  - Enhanced internationalization by adding new translation properties for improved localization support.

### CODE REVIEW

**Comment by Justin-Avaya** [comment]
**Issue 01**

I find the following both misleading and confusing. 

1. Select either an item, multiple items or all items on a page or even multiple items across multiple pages.
2. Open dropdown menu
   - Clear all (10 items)

// To me this feels like "10 items" are currently selected, which is not the case. In the example attached below a single item is selected.

Possible solution(s):
Clear all
_or_
Clear all (1 selected item)

See example:
Screenshot 2024-07-19 at 12 33 52 PM

<hr>

**Issue 02**
(possibly out of scope)

Tricky hover treatment. Hovering over the left area of the dropdown target doesn't seem to trigger a hover event.
As soon as the mouse enters the hover target area, the hover treatment is displayed including the area that was not initially tripping the hover target.

Possible solution:
Allow the entire area to receive the hover treatment.

See example:


<hr>

**Issue 03**
(possibly out of scope)

Blue banner text for both "Select all items in the table." and "Clear all items in the table." - I have a problem with the term "Table." I don't think of table as meaning _'all items on all pages'_ rather, I think of a table as only what i see displayed. Since we are using the term "Page" elsewhere in dropdown menus, I suggest we use similar language. 

Possible solution(s):
With at least 1 but not all items selected: "Select all items on all pages"
With all items selected: "Clear all items on all pages"

See examples:
Screenshot 2024-07-19 at 2 42 27 PM
Screenshot 2024-07-19 at 2 39 02 PM

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 1]
```diff
@@ -144,44 +151,50 @@ export const TableHeader = <T extends Record<string, any>>({
...
-										{allPageEnabledRowsSelected ? (
-											<>
-												{translations.clearPage} ({pageEnabledRowCount})
-											</>
-										) : (
-											<>
-												{translations.selectPage} ({pageEnabledRowCount})
-											</>
-										)}
+										{translations.selectAll} ({tableEnabledRowCount}{" "}
```
Comment (enrique-prado) [inline src/components/Table/TableComponents/TableHeader.tsx:186]: There is a weird {" "} at the end of this line. I've seen the IDE add these things. Should probably remove.

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 2]
```diff
@@ -144,44 +151,50 @@ export const TableHeader = <T extends Record<string, any>>({
...
-										{allTableEnabledRowsAreSelected ? (
-											<>
-												{translations.clearAll} ({tableEnabledRowCount})
-											</>
-										) : (
-											<>
-												{translations.selectAll} ({tableEnabledRowCount})
-											</>
-										)}
+										{translations.clearAll} ({tableEnabledRowCount}{" "}
```
Comment (enrique-prado) [inline src/components/Table/TableComponents/TableHeader.tsx:196]: same thing here.

File: src/components/Table/helpers/default-data.ts [hunk 3]
```diff
@@ -20,10 +20,11 @@ export const translations: ITableTranslations = {
 	header: {
 		clearSort: "Clear Sort",
 		filterColumn: "Filter Column",
-		clearAll: "Clear all items in the table",
-		selectAll: "Select all items in the table",
-		clearPage: "Clear all items on page",
-		selectPage: "Select all items on page",
+		clearAll: "Clear all",
```
Comment (enrique-prado) [inline src/components/Table/helpers/default-data.ts:23]: Nice. I like these shorter strings better.

**Review by enrique-prado — APPROVED** [review]
Nice work. Just the 2 comments I made.

**Review by yangAtSpoken — APPROVED** [review]

**Comment by joe-s-avaya** [comment]
@Justin-Avaya 

> **Issue 01**

Per discussion in stand-up, I am waiting to hear from you or Matt on the final wording.


> **Issue 02** (possibly out of scope)

This was brought in by the dnd work (by Yang). This is out of scope for this PR. Would you like me to create a bug for this? 


> **Issue 03** (possibly out of scope)

Per discussion in stand-up, I am waiting to hear from you or Matt on the final wording.

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableHeader.tsx:186]: If there isn't a space the wording runs up against the number. Example without the `{" "}`:
Select all (5items)

File: src/components/Table/TableComponents/TableHeader.tsx (hunk 2 above)
Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableHeader.tsx:196]: see other conversation, resolving this one

**Comment by Justin-Avaya** [comment]
> @Justin-Avaya
> 
> > **Issue 01**
> 
> Per discussion in stand-up, I am waiting to hear from you or Matt on the final wording.
> 
> > **Issue 02** (possibly out of scope)
> 
> This was brought in by the dnd work (by Yang). This is out of scope for this PR. Would you like me to create a bug for this?
> 
> > **Issue 03** (possibly out of scope)
> 
> Per discussion in stand-up, I am waiting to hear from you or Matt on the final wording.

@joe-s-avaya 
Matt and I have met and discussed both Issue 01 and 03. We've crafted wording, submitted to David (UX copy) which was approved. The wording will be presented to Jenn for her approval (I believe tomorrow)

**Comment by joe-s-avaya** [comment]
@matthewAvaya and @Justin-Avaya based on the conversation we had in standup today, can I merge this as is and then create another jira issue once we have new wording/UX?

**Review by Justin-Avaya — APPROVED** [review]

**Comment by joe-s-avaya** [comment]
Matt has updated the AC and this work is deprecated

**Pull request closed without merging**